src/engine/SpatialJoinAlgorithms.cpp

```diff
@@ -629,6 +629,28 @@ double SpatialJoinAlgorithms::getMaxDistFromMidpointToAnyPointInsideTheBox(
 
 // ____________________________________________________________________________
 Result SpatialJoinAlgorithms::BoundingBoxAlgorithm() {
+  auto getRtreeEntry = [&](const IdTable* idTable, const size_t row,
+                           const ColumnIndex col) -> std::optional<rtreeEntry> {
+    rtreeEntry entry{row, std::nullopt, std::nullopt, std::nullopt};
+    entry.geoPoint_ = getPoint(idTable, row, col);
+
+    if (!entry.geoPoint_) {
+      entry.geometry_ = getAnyGeometry(idTable, row, col);
+      if (!entry.geometry_.has_value()) {
+        return std::nullopt;
+      }
+      entry.boundingBox_ =
+          boost::apply_visitor(BoundingBoxVisitor(), entry.geometry_.value());
+    } else {
+      entry.boundingBox_ =
+          Box(Point(entry.geoPoint_.value().getLng(),
+                    entry.geoPoint_.value().getLat()),
+              Point(entry.geoPoint_.value().getLng() + 0.00000001,
+                    entry.geoPoint_.value().getLat() + 0.00000001));
+    }
+    return entry;
+  };
```

Comment:
I like the refactoring, but as this lambda is 
1. rather long and 2. doesn't capture anything, you can also make it a separate function (probalby a member function then).